tests/baselines/reference/genericMappedTypeAcceptsInferredObjectType.errors.txt

```diff
@@ -0,0 +1,50 @@
+tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts(27,14): error TS2345: Argument of type '{ foo: T["foo"]; }' is not assignable to parameter of type 'GenericMap<T>'.
+tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts(28,14): error TS2345: Argument of type '{ bar: T["bar"]; }' is not assignable to parameter of type 'GenericMap<T>'.
+tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts(31,14): error TS2345: Argument of type '{ foo: T["foo"]; }' is not assignable to parameter of type 'OptionalGenericMap<T>'.
+tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts(32,14): error TS2345: Argument of type '{ bar: T["bar"]; }' is not assignable to parameter of type 'OptionalGenericMap<T>'.
+
+
+==== tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts (4 errors) ====
+    interface Constraint {
+    	foo: string
+    }
+    
+    interface ExtendedConstraint extends Constraint {
+    	bar: string
+    }
+    
+    type GenericMap<T> = {
+    	[P in keyof T]: T[P]
+    }
+    
+    type OptionalGenericMap<T> = {
+    	[P in keyof T]?: T[P]
+    }
+    
+    const required = <T>(x: GenericMap<T>) => x
+    const optional = <T>(x: OptionalGenericMap<T>) => x
+    
+    const withinConstraint = <T extends Constraint>(foo: T['foo']) => {
+    	required<T>({ foo }) // no error as { foo: T['foo'] } <: GenericMap<T>

```

Comment:
This is incorrect, tsc should reject this call. `T` could be `{foo: string, bar: number}`. Then `{foo: string}` should definitely not be assignable to `{foo: string, bar: number}`.

It's not the case that `{foo: T['foo'] }` is definitely a subtype of `GenericMap<T>`. It's only definitely a subtype of `OptionalGenericMap<T>` (that that in TS's type system today; it is possible that this could change in the future).

This was the meaning of my [original comment](https://github.com/microsoft/TypeScript/issues/37670#issuecomment-634400539) on #37670 ; tsc _should reject_ this code, that's not a defect in tsc.

Only `optional` calls below are fine.

Reply:
Yes, you're absolutely right! I did overlook that.